src/main/java/eu/merloteducation/organisationsorchestrator/controller/OrganizationQueryControllerAdvice.java

```diff
@@ -41,6 +41,11 @@ protected void beforeBodyWriteInternal(MappingJacksonValue bodyContainer, @NotNu
                     !authorityChecker.representsOrganization(authentication, participantDto.getId())) {
                 participantDto.setMetadata(null); // for single objects hide metadata if we are not representing
             }
+
+            if (participantDto.getMetadata() != null && !authorityChecker.representsOrganization(authentication, participantDto.getId())) {
```

Comment:
maybe extractthe representsOrganization check and save it in a boolean to avoid multiple calls, same below for the page